Signal/src/ViewControllers/BlockListViewController.m

```diff
@@ -214,6 +235,35 @@ - (void)blockedPhoneNumbersDidChange:(id)notification
     [self.tableView reloadData];
 }
 
+- (void)signalRecipientsDidChange:(NSNotification *)notification
+{
+    [self updateContacts];
+}
+
+- (void)updateContacts
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        self.contacts = self.contactsManager.signalContacts;
+        [self.tableView reloadData];
+    });
+}
+
+- (void)setContacts:(NSArray<Contact *> *)contacts
+{
+    _contacts = contacts;
+
+    NSMutableDictionary<NSString *, Contact *> *contactMap = [NSMutableDictionary new];
```

Comment:
nit: `self.contactMap` seems to be re-implementing `[contactManager contactForPhoneIdentifier:]` (but more efficiently!). Would you consider modifying/using that?